src/OpenTelemetry.Api/AssemblyInfo.cs

```diff
@@ -16,6 +16,7 @@
 
 using System.Runtime.CompilerServices;
 
+[assembly: InternalsVisibleTo("OpenTelemetry" + AssemblyInfo.PublicKey)]
```

Comment:
which particular class/api is triggering this change? can we keep using shared file approach instead?

Reply:
The `PeerServiceResolver` I refactored into SDK needs SemanticConventions, which is in API. So I included that file in SDK. Problem is, OpenTelemetry.Tests won't build with that. Because OpenTelemetry.Tests can see the internals of API & SDK, it was getting an error like "SemanticConventions type exists in both API & SDK" 🤯 I couldn't figure out a more elegant way to fix it. Open to ideas!

Reply:
FYI - I've seen [this workaround](https://github.com/open-telemetry/opentelemetry-dotnet/blob/296483c2d31043ee517c45adf124506f0d29475d/src/OpenTelemetry.Api/Context/Propagation/TraceStateUtilsNew.cs#L109). Not sure if I would call it "elegant" though 🧙‍♀️

Reply:
Okay got it. I am okay to keep it like this.

Reply:
@reyang I think I actually removed that when I made this change 🤣 But I could put that back and just make sure API & SDK SemanticConventions each get a different namespace, that would fix it. Not sure which approach is better. Why I went with this one: This is letting SDK see everything in API. SDK is kind of a superset of API so it really should see everything? 🤷 

PS: One other change that is easy to miss on the diff: In the projects I touched, I move all the includes to /Includes/ folder. It was a little inconsistent before, some in root, some in /Internal/, some in /Implementation/.

Reply:
@CodeBlanch sorry I didn't walk through this PR.
I would vote for making the API internals visible to the SDK since the namespace hack is hard to maintain/debug/test.

Reply:
One thing we might need to consider is the API/SDK version compat.
E.g. do we expect version 1.1 of the SDK to work with 1.0 API?